plugins/inputs/dpdk/dpdk.go

```diff
@@ -96,136 +82,236 @@ func (dpdk *dpdk) Init() error {
+	for _, socketToDisconnect := range socketsToDisconnect {
+		for i := 0; i < len(dpdk.connectors); i++ {
+			connector := dpdk.connectors[i]
+			if socketToDisconnect == connector.pathToSocket {
+				dpdk.Log.Debugf("Close unused connection: %s", socketToDisconnect)
+				if closeErr := connector.tryClose(); closeErr != nil {
+					dpdk.Log.Warnf("Failed to close unused connection - %v", closeErr)
+				}
+				dpdk.connectors = append(dpdk.connectors[:i], dpdk.connectors[i+1:]...)
+				i--
+				break
+			}
+		}
+	}
+
+	// Create connections
+	for _, socketToConnect := range socketsToConnect {
+		connector := newDpdkConnector(socketToConnect, dpdk.AccessTimeout)
+		connectionInitMessage, err := connector.connect()
+		if err != nil {
+			if dpdk.UnreachableSocketBehavior == unreachableSocketBehaviorError {
+				return fmt.Errorf("couldn't connect to socket %s: %w", socketToConnect, err)
+			}
+			dpdk.Log.Warnf("Couldn't connect to socket %s: %v", socketToConnect, err)
+			continue
+		}
+
+		dpdk.Log.Debugf("Successfully connected to the socket: %s. Version: %v running as process with PID %v with len %v",
+			socketToConnect, connectionInitMessage.Version, connectionInitMessage.Pid, connectionInitMessage.MaxOutputLen)
+		dpdk.connectors = append(dpdk.connectors, connector)
+	}
 
-	for _, command := range commands {
-		dpdk.processCommand(acc, command)
+	if len(dpdk.connectors) == 0 {
+		return fmt.Errorf("no active sockets connections present")
 	}
 
 	return nil
 }
 
+// Identify scope of sockets to connect/disconnect
+func (dpdk *dpdk) identifySockets() (socketsToConnect []string, socketsToDisconnect []string) {
+	candidates := []string{dpdk.SocketPath}
+	if choice.Contains(dpdkPluginOptionInMemory, dpdk.PluginOptions) {
+		candidates = dpdk.getDpdkInMemorySocketPaths()
+	}
+
+	// Find sockets in the connected-sockets list that are not among
+	// the candidates anymore and thus need to be removed.
+	toDisconnect := make([]string, 0, len(dpdk.connectors))
+	for _, connector := range dpdk.connectors {
+		var found bool
+		for _, candidate := range candidates {
+			if candidate == connector.pathToSocket {
+				found = true
+				break
+			}
+		}
+		if !found {
+			toDisconnect = append(toDisconnect, connector.pathToSocket)
+		}
+	}
+
+	// Find candidates that are not yet in the connected-sockets list as we
+	// need to connect to those later.
+	toConnect := make([]string, 0, len(candidates))
+	for _, candidate := range candidates {
+		var found bool
+		for _, connector := range dpdk.connectors {
+			if candidate == connector.pathToSocket {
+				found = true
+				break
+			}
+		}
+		if !found {
+			toConnect = append(toConnect, candidate)
+		}
+	}
+
+	return toConnect, toDisconnect
+}
+
 // Gathers all unique commands
-func (dpdk *dpdk) gatherCommands(acc telegraf.Accumulator) []string {
+func (dpdk *dpdk) gatherCommands(dpdkConnector *dpdkConnector, acc telegraf.Accumulator) []string {
```

Comment:
We usually do have `acc` as the first parameter. Can you please change the order?

Reply:
Addressed in: https://github.com/influxdata/telegraf/pull/14308/commits/594fecb38d9f68e33ff03a50172ee132bc5811f6